examples/images_contours_and_fields/pcolormesh_grids.py

```diff
@@ -0,0 +1,141 @@
+"""
+============================
+Pcolormesh grids and shading
+============================
+
+`.axes.Axes.pcolormesh` and `~.axes.Axes.pcolor` have a few options for
+how grids are laid out and the shading between the grid points.
+
+Generally, if *Z* is size *MxN* then the grid *X* and *Y* can be
+specified with either size *M+1 x N+1* or *MxN*, depending on the
+argument for the ``shading`` keyword argument.
+
+"""
+
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import BoundaryNorm
+from matplotlib.ticker import MaxNLocator
+import numpy as np
+
+###############################################################################
+# Flat Shading
+# ------------
+#
+# The grid specification with the least assumptions is ``shading='flat'``
+# and if the grid is one larger than the data in each dimesion, i.e. has size
+# *M+1 x N+1*.  In that case *X* and *Y* sepcify the corners of quadrilaterals
+# that are colored with the values in *Z*:
+#
+#        (X[i+1, j], Y[i+1, j])       (X[i+1, j+1], Y[i+1, j+1])
+#                          +-----+
+#                          |     |
+#                          +-----+
+#        (X[i, j], Y[i, j])       (X[i, j+1], Y[i, j+1])
+#
+# Here we specify the edges of the *3x5* quadirlaterals with *X* and *Y* that
+# are *4x6*.
+np.random.seed(19680801)
+Z = np.random.rand(3, 5)
+x = np.arange(6)  # len = 6
+y = np.arange(4)  # len = 4
```

Comment:
I didn’t do this because I found having arbitrary pattern a bit easier to see what’s cut and what’s not.  But I can play w it and see. Maybe I’m wrong and something analytical would be better

Reply:
Oh and I guess the other reason is that z(x,y) really favours the nearest interpretation.  Ie centring the data there is unambiguously correct.  To make it correct for shading=flat you’d need to choose x,y in the middle of the grid cell. Here I wanted z to be the same for every example but not definitively correct for any particular example.

Reply:
Sounds good to me. I agree that this probably describes all of the uses a little more generically.

Reply:
I changed it to `np.arange(15).reshape(3, 5)`.  Thats explicit, and clearly shows when things have been dropped, so maybe a good compromise?